app/services/grade_modules.rb

```diff
@@ -167,4 +188,34 @@ def grade_assignment(canvas_assignment_id, user_ids)
       ).where('id <= ?', max_id).update_all(new: false)
     end
   end
+
+  def self.grading_disabled_for?(canvas_course_id, canvas_assignment_id, user)
+    rise360_module_grade = CourseRise360ModuleVersion.find_by!(canvas_assignment_id: canvas_assignment_id)
+      .rise360_module_grades.find_by(user: user)
+
+    # If they've never opened the module a Rise360ModuleGrade won't exist yet.
+    # All auto-grading logic is disabled circuited until they do.
+    unless rise360_module_grade
+      Honeycomb.add_field('grade_modules.module_never_opened', true)
+      return true
+    end
```

Comment:
I think something like this would be easier to read?
```suggestion
    Honeycomb.add_field('grade_modules.module_opened', rise360_module_grade.present?)
    return true unless rise360_module_grade
```